googledataprocauthenticator/controllerwidget/addendpointwidget.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Creates the widget under the Sessions tab within the ``%manage_dataproc widget``"""
+
+import importlib
+from sparkmagic.livyclientlib.endpoint import Endpoint
+import sparkmagic.utils.configuration as conf
+from sparkmagic.controllerwidget.abstractmenuwidget import AbstractMenuWidget
+import ipyvuetify as v
+from googledataprocauthenticator.utils.utils import SerializableEndpoint, get_stored_endpoints
+from googledataprocauthenticator.utils.constants import WIDGET_WIDTH
+
+
+class AddEndpointWidget(AbstractMenuWidget):
+
+    def __init__(self, spark_controller, ipywidget_factory, ipython_display, endpoints,
+                 refresh_method, state, db):
+        super(AddEndpointWidget, self).__init__(
+            spark_controller, ipywidget_factory, ipython_display, True)
+        self.endpoints = endpoints
+        self.refresh_method = refresh_method
+        self.state = state
+        self.delete_pressed = False
+        self.db = db
+
+        auth = conf.authenticators().get("Google")
```

Comment:
Let's just direct import the Google auth class here, since this isn't likely to change.